Thumbnail changes

georgeteo opened the pull request:
@fatlotus @tbkissinger 

**Don't merge yet**

I thought I would spend some time tinkering with the thumbnails because I came across a site that I liked. 

What do you think about this new version: https://thumbnail-changes-dot-hosted-caravel.appspot.com/

The category tag at the top is a filler for now. I am thinking of having color coded category tags.

fatlotus commented:
Can't see it (looks like too many versions used up our CPU quota for the day); will look at it tomorrow.

fatlotus commented on caravel/static/default.css:
```diff
@@ -59,7 +59,7 @@ body {
     right: 15px;
     padding: 5px 10px;
-    font: bold 1.5em/30px Roboto,sans-serif;
+    font: 300 1.5em/30px Roboto,sans-serif;
```
Er, didn't we already discuss this?

georgeteo replied:
We did. I just wanted to see how it looks with a thinner font. Your thoughts on the overall design?

fatlotus commented on caravel/static/default.js:
```diff
@@ -0,0 +1,14 @@
+var mq = window.matchMedia('all and (max-width: 700px)');
```
Why is this here?

georgeteo replied:
This is not necessary. This branch is not ready to merge. I just wanted to show you the UI.

fatlotus commented on caravel/templates/index.html:
```diff
@@ -1,28 +1,43 @@
+    {%  if i%4 == 0 %}
+      <div class="row row-eq-height">
+    {% endif %}
+    <div class="col-xs-12 col-md-3" style="margin-left: 10px; margin-right: 10px; margin-bottom:15px; padding:0; background-color: #ffffff;
```
???

georgeteo replied:
Fixing constant height windows. But requires using the row div for every 4 cols. 

Planed to add or remove those tags based on responsible JS. Hence the default.js file.

fatlotus commented:
While it's good to experiment, this branch feels to me like a science experiment. What exactly are you trying to accomplish here?

georgeteo commented:
This branch is a science experiment. I just wanted to fiddle around with the listings page to see if I could find something that I liked and this is what I ended up with. This is not meant to be merged in unless you happen to like all of it too. 

Some things that I changed: 
1. Fixed size col height. 
2. Title and price below the image
3. Round tags for categories (that can be color coded).
4. New no image uploaded picture. 
5. Thinner fonts

fatlotus commented:
2. Why de-emphasize category and posting time? Presumably, most people will be browsing by category, and if not, the category (should be) self-evident from the category and/or picture.

4. I think the "no image uploaded" picture is a little distracting. Might it make more sense to have a default coded by the category of the image? I was thinking of using The Noun Project images, since they're matte and often public domain.

6. Are you sure we're looking at the same version? At present (coped from `hosted-caravel.appspot.com`), I think we have:

```
<div class="listing col-xs-12 col-md-3">
  <h3 class="title">
    <a href="/baby-play-yard?q=">Baby Play Yard</a>
  </h3>
  <p class="price">$60.00</p>
  <p class="age">4d</p>
  <a href="/baby-play-yard?q=">
    <img src="blahblah" class="panel-image-preview">
  </a>
</div>
```

georgeteo commented:
<img width="589" alt="screen shot 2015-09-19 at 9 10 56 pm" src="https://cloud.githubusercontent.com/assets/12107160/9979185/2881e4fe-5f13-11e5-872c-617cba813d69.png">

georgeteo commented:
Can we come to a verdict about this?

Could also ask Tyler to weigh in if we can't come to a decision.

fatlotus commented:
What about my other comments above? Specifically, the default image and "cleaner" markup?

I still don't really understand why we need to separately label with categories, though I could be convinced of moving the prices closer to the titles.

georgeteo commented:
Re comments above:

(2) 

(4) I am open to other images there. Please suggest something else. 

(6) I am looking at the ones in thumbnail-changes-dot-hosted-caravel.appspot.com. 

```
<div class="col-xs-12 col-md-3" style="margin-left: 10px; margin-right: 10px; margin-bottom:15px; padding:0; background-color: #ffffff;
    border-radius: 10px; border: 1px solid #e5e5e5">
      <div style="position: absolute; left: 10px; top: 10px; background: #900; padding: 3px; border-radius: 5px; opacity: 0.8;">
        <span style="color: white;"> Category </span>
      </div>



      <a href="/627fce04-b3e3-4e4b-8420-eb8b1ae70ee4?q=">
        
          <img style="width:100%;" src="/static/no-image.png">
        
      </a>
      <h3 style="background-color: #ffffff">
        <p style="padding-left: 10px;">
          <a href="/627fce04-b3e3-4e4b-8420-eb8b1ae70ee4?q=" style="font-size: 15px; font-weight:400; color: black;">
            Hello
          </a>
        </p>
        <p style="font-weight:300; padding-left:10px; font-size:15px"> $4,242.00</p>
      </h3>
    </div>
```

georgeteo commented:
bump. replies?

fatlotus commented:
Sorry, thought I already responded to this.

The site (The Gadget Flow) looks nice. It's not really the aesthetic I had in mind, but at this point I just want to ship this thing.

For (4), I believe I already suggested taking a look at the The Noun Project, which provides vector iconography for pretty much anything; we'd probably grab light grey versions of "house," "car," and "apartment" nouns, somehow connecting with the categories view, once it's added.

For (6), I disagree that the markup you've proposed is cleaner. As it stands, the title is at the bottom, which will make the site less accessible to screen readers. I think it's possible with flex box and/or absolute positioning to keep it at the top, though. (Also inline styles, but I realize this is for development.)

For the moment, can we focus on getting it working? I feel like there's enough v.1 blockers that we should probably defer styling changes until we have users.

georgeteo commented:
Okay. Will focus on getting other things working, but I am worried that if we launch the site with the current UI we might lose users. 

But alright, lets table this for now.

fatlotus commented:
You know, what the hell, let's do it. Send me something without inline styles and I'll merge.

georgeteo commented:
Ok have uninlined css.

A few weird issues:
1. I have added a javascript handler for removing the <div class="row row-eq-height"> divs that need to be put in every 4 <div class="col-md-3"> for small sceens. It works on a local dev sever, but doesn't seem to be working on the app engine version. Do you know why?

2. Seemed to have broke the show page when I merged the master branch and fixed merge conflicts. Also any idea why?

fatlotus commented:
1. Yuck. I really think this should be solved by having equal height boxes — adding `class="row"`, or even dynamic JavaScript, sounds like a recipe for a bad day.
2. Not sure — what do you mean by "broke"?

georgeteo commented:
1. Unless we want to truncate title names will be hard to have equal size boxes.  

2. Try clicking through to a listing. The query seems to be wonky.

fatlotus commented on caravel/templates/listings/inline.html:
```diff
@@ -1,20 +1,22 @@
+  <div class="category-tag">
+    <span class="category-text"> Category </span>
+  </div>
+  <a href="/{{ listing.key }}?q={{ request.args['q']}}">
```
Use `url_for("show_listing", permalink=listing.permalink, q=request.args['q']` (my bad).

fatlotus commented on caravel/templates/listings/inline.html:
```diff
@@ -1,20 +1,22 @@
\ No newline at end of file
+  <h3 class="thumbnail-title-block">
+    <p class = "thumbnail-title">
+      <a href="/{{ listing.key }}?q={{ request.args['q']}}" class="thumbnail-title">
```
Same.

fatlotus commented:
IIRC, we used to truncate the image in order to fit the full title, then truncate the title once there was no image left. (There was a bug, though, where the text placeholder in a wide window was not the right height — that should be fixed by using an image placeholder.). Is there a way to keep that behavior, sans placeholder bug? The rivers make it feel a little uneven.

fatlotus commented:
Sorry, maybe I was unclear in my last comment. I mean, the sizing bug will be fixed (mostly) if we do two things:
1. Switch from a textual placeholder to an image, as you've done.
2. Make the title expand over the listing image.

Once we do that, we don't need extra JavaScript or non-semantic markup (https://en.wikipedia.org/wiki/Semantic_HTML).

georgeteo commented:
Okay. Have gotten rid of the JS, but there is a little bug. It is mostly the same height, but without the row divs, it isn't perfect. 

Take a look at https://thumbnail-changes-dot-hosted-caravel.appspot.com/. The rows that are weird are because the row on top has some thumbnails that are 316px and some that are 317 px, but I can't fix the height of each thumbnail, because the height would be fixed for phones as well and that would be weird. I could make that one CSS class responsive, but am wondering if there is a better fix.

georgeteo commented:
Wohhooo...Fixed! Take a look again.

Btw, could you try looking at the site with display width ~750/760. There is this weird point somewhere around there where the vertical thumbnails only partially kicks in, so the image is visible but the text is not.

fatlotus commented:
Looks good to me! I can't see the bug you're describing, so feel free to merge when you think it looks fixed.

fatlotus commented:
oh dear

georgeteo commented:
Oh dear? looks good to me?

fatlotus commented on caravel/templates/listings/inline.html:
```diff
@@ -1,20 +1,24 @@
\ No newline at end of file
+  <h3 class="thumbnail-title-block">
+    <p class="thumbnail-title">
+      <a href="url_for('show_listing', permalink=listing.permalink, q=request.args['q'])" class="thumbnail-title">
```
This broke stuff.

georgeteo replied:
I know. Will fix will new pull request in discrete.
